File: obdii/viewmodels/BaseViewModel.swift
/**
 
 * __Final Project__
 * Jim Mittler
 * 19 November 2025
 
 
Base Class for View Models so all the ViewModels  have  a standard onChanged hook.
 
 We use this hook on the CarPlay side to trigger refreshes because CarPlay doesn't support the full two-way binding like SwiftUI
 
 _Italic text__
 __Bold text__
 ~~Strikethrough text~~
 
 */
@MainActor
class BaseViewModel {
    var onChanged: (() -> Void)?
}

File: flutter_obdii/lib/viewmodels/base_view_model.dart
// Port of BaseViewModel.swift — Jim Mittler
// Base class for all ViewModels.
// Provides a standard onChanged hook so all ViewModels can trigger
// CarPlay refreshes (CarPlay doesn't support two-way SwiftUI binding).

import 'package:flutter/foundation.dart';

abstract class BaseViewModel extends ChangeNotifier {
  /// Hook for CarPlay integration — set by the CarPlay controller to
  /// receive notifications when data changes.
  VoidCallback? onChanged;

  @override
  void notifyListeners() {
    super.notifyListeners();
    onChanged?.call();
  }
}
